client/matchmaking/admin-matchmaking-times.tsx

```diff
@@ -0,0 +1,690 @@
+          setLastError(err)
+        },
+      }),
+    )
+
+    return () => abortController.abort()
+  }, [activeTab, dispatch, setCurrentTime, setLastError])
+
+  if (isLoadingCurrentTime) {
+    return <LoadingDotsArea />
+  }
+
+  if (lastError) {
+    return <ErrorText>Error retrieving current matchmaking time: {lastError.message}</ErrorText>
+  }
+
+  if (!currentTime) {
+    return (
+      <EmptyStateContainer>
+        <EmptyStateIcon icon='schedule' />
+        <EmptyStateText>
+          No matchmaking times have been scheduled for this matchmaking type yet.
+        </EmptyStateText>
+      </EmptyStateContainer>
+    )
+  }
+
+  return (
+    <>
+      <SectionTitle>Current Matchmaking Time</SectionTitle>
+      <CurrentMatchmakingCard>
+        <HistoryInfo>
+          <HistoryDate>
+            {longTimestamp.format(currentTime.startDate)} <CurrentText>(Current)</CurrentText>
+          </HistoryDate>
+          <HistoryStatus>
+            <MatchmakingStatus isEnabled={currentTime.enabled} />
+          </HistoryStatus>
+        </HistoryInfo>
+      </CurrentMatchmakingCard>
+    </>
+  )
+}
+
+function PastMatchmakingTimes({ activeTab }: { activeTab: MatchmakingType }) {
+  const dispatch = useAppDispatch()
+
+  const [pastTimesIds, setPastTimesIds] = useImmerState<Set<number>>(new Set())
+  const [pastTimesById, setPastTimesById] = useImmerState<Map<number, MatchmakingTimeJson>>(
+    new Map(),
+  )
+
+  const [hasMorePastTimes, setHasMorePastTimes] = useState(true)
+  const [isLoadingPastTimes, setIsLoadingPastTimes] = useState(false)
+  const [lastError, setLastError] = useState<Error | undefined>(undefined)
+
+  const onLoadMorePastTimes = useCallback(
+    (offset: number) => {
+      setIsLoadingPastTimes(true)
+
+      dispatch(
+        getMatchmakingTimesPast(activeTab, offset, {
+          onSuccess: data => {
+            data.pastTimes.forEach(time => {
+              setPastTimesIds(draft => {
+                draft.add(time.id)
+              })
+              setPastTimesById(draft => {
+                draft.set(time.id, time)
+              })
+            })
+            setHasMorePastTimes(data.hasMorePastTimes)
+            setIsLoadingPastTimes(false)
+            setLastError(undefined)
+          },
+          onError: err => {
+            setIsLoadingPastTimes(false)
+            setLastError(err)
+          },
+        }),
+      )
+    },
+    [activeTab, dispatch, setPastTimesById, setPastTimesIds],
+  )
+
+  useEffect(() => {
+    const abortController = new AbortController()
+
+    onLoadMorePastTimes(0)
+
+    return () => abortController.abort()
+  }, [onLoadMorePastTimes])
+
+  if (lastError) {
+    return <ErrorText>Error retrieving past matchmaking times: {lastError.message}</ErrorText>
+  }
+
+  if (pastTimesIds.size === 0) {
+    return null
+  }
+
+  return (
+    <>
+      <SectionTitle>Past Matchmaking Times</SectionTitle>
+      {Array.from(pastTimesIds).map(id => {
```

Comment:
`Array.from`'s second parameter is a `map` function, use it instead of calling `map`  after